packages/client-sdk-nodejs/src/config/middleware/experimental-metrics-logging-middleware.ts

```diff
@@ -0,0 +1,47 @@
+import {MomentoLogger, MomentoLoggerFactory} from '../../';
+import {
+  ExperimentalMetricsMiddleware,
+  ExperimentalMetricsMiddlewareRequestHandler,
+  ExperimentalRequestMetrics,
+} from './impl/experimental-metrics-middleware';
+
+class ExperimentalMetricsLoggingMiddlewareRequestHandler extends ExperimentalMetricsMiddlewareRequestHandler {
+  constructor(parent: ExperimentalMetricsMiddleware, logger: MomentoLogger) {
+    super(parent, logger);
+  }
+
+  emitMetrics(metrics: ExperimentalRequestMetrics): Promise<void> {
+    this.logger.info(JSON.stringify(metrics));
+    return Promise.resolve();
+  }
+}
+
+/**
+ * This middleware enables per-request client-side metrics.  Metrics for each
+ * request will be written to a CSV file; this file can be analyzed or shared
+ * with Momento to diagnose performance issues.
+ *
+ * The metrics format is currently considered experimental; in a future release,
+ * once the format is considered stable, this class will be renamed to remove
+ * the Experimental prefix.
+ *
+ * WARNING: enabling this middleware may have minor performance implications,
+ * so enable with caution.
+ *
+ * WARNING: depending on your request volume, the CSV file size may grow quickly;
+ * neither sampling nor file compression / rotation are included at this time
+ * (though they may be added in the future).
```

Comment:
This makes me think that very few people will be comfortable enbaling this in their production workflows

Reply:
these comments are wrong, this implementation doesn't actually write to a CSV file. will fix.

Reply:
added a commit to fix this comment in https://github.com/momentohq/client-sdk-javascript/commit/53e734bc7334d4275dee7010c4a05afe747774d2 .